python/lsst/ts/mtdomegui/model.py

```diff
@@ -62,11 +125,605 @@ def __init__(
 
         # TODO: Put this into the ts_config_mttcs in DM-48109.
         self.connection_information = {
-            "host": "localhost",
-            "port": 1000,
-            "timeout_connection": 10.0,
+            "host": host,
+            "port": port,
         }
 
         self.duration_refresh = duration_refresh
 
         self.reporter = Reporter(self.log)
+
+        self.mtdome_com: MTDomeCom | None = None
+
+        self.periodic_tasks: list[asyncio.Task] = list()
+
+    async def connect(self) -> None:
+        """Connect to the controller.
+
+        Start the mock controller, if simulating.
+        """
+
+        config = type(
+            "Obj",
+            (object,),
+            {key: value for key, value in self.connection_information.items()},
+        )()
+        self.mtdome_com = MTDomeCom(
+            self.log, config, simulation_mode=self._is_simulation_mode
+        )
+
+        try:
+            await self.mtdome_com.connect()
+        except Exception as error:
+            raise RuntimeError(f"Connection to the controller failed: {error!r}.")
+
+        self.reporter.report_state_azimuth_axis(MTDome.EnabledState.ENABLED)
+        self.reporter.report_fault_code_azimuth_axis("")
+
+        self.reporter.report_state_elevation_axis(MTDome.EnabledState.ENABLED)
+        self.reporter.report_fault_code_elevation_axis("")
+
+        self.reporter.report_state_aperture_shutter(MTDome.EnabledState.ENABLED)
+        self.reporter.report_fault_code_aperture_shutter("")
+
+        self.reporter.report_state_brake_engaged(0)
+        self.reporter.report_state_locking_pins_engaged(0)
+
+        self.reporter.report_state_power_mode(self.mtdome_com.power_management_mode)
+
+        # Start polling for the status of the lower level components
+        # periodically.
+        await self.start_periodic_tasks()
+
+        self.log.info("connected")
+
+    async def start_periodic_tasks(self) -> None:
```

Comment:
This is an exact copy of the method in ts_mtdome. We should extract the one from ts_mtdome and put it in ts_mtdomecom.

Reply:
Done in the commit: [17d71c3](https://github.com/lsst-ts/ts_mtdomegui/pull/4/commits/17d71c31a9d25a477c45cf52d78acfc78b39afce). Thanks!